src/main/java/com/dakar/dakar/controller/RequestHandler.java

```diff
@@ -0,0 +1,37 @@
+package com.dakar.dakar.controller;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import javax.validation.Validator;
+import java.awt.print.Book;
+import java.net.URI;
+import java.util.function.Function;
+
+/**
+ * Handler for requests
+ */
+@Component
+public class RequestHandler {
+
+    private final Validator validator;
+
+    public RequestHandler(Validator validator) {
+        this.validator = validator;
+    }
+
+    public <BODY> Mono<ServerResponse> requireValidBody(
+            Function<Mono<BODY>, Mono<ServerResponse>> block,
+            ServerRequest request, Class<BODY> bodyClass) {
+
+        return request
+                .bodyToMono(bodyClass)
+                .flatMap(
+                        body -> validator.validate(body).isEmpty()
```

Comment:
looks like it's the same treatment that you're doing in this endpoint : saveJourneyValidatorJavax